src/clr-core/common/utils/aria-live.service.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { AriaLivePoliteness } from '../enums/aria-live-politeness';
+import { applyMixins } from '../mixins/apply-mixins';
+import { UniqueId } from '../mixins/unique-id';
+
+/**
+ * Time in milliseconds before inserting the content into the container
+ */
+export const ARIA_LIVE_TICK: number = 100;
+export class CwcAriaLiveService {
+  protected previousTimeout: ReturnType<typeof setTimeout>;
+  protected ariaLiveElement: HTMLElement | null;
+  document: Document;
+
+  readonly id = `clr-aria-live-element-${this._uniqueId}`;
```

Comment:
I’m looking at this from my phone .. but I don’t see from where the `_uniqueId ` Is coming ? Is applyMixing the one that is creating it? Just asking ?

Reply:
The uniqueId is added via a typescript mixin. We have several of them in core that we can reuse across different components. See ln 124 of this file for where the mixin is applied.

Reply:
Ok, so if this is really uniq and we make new instance based on component creation, this service will work as expected. But there is room for improvements. 

But whatever we change here it will be nice to make the changes into the angular service or go with a single service that is working on both packages.

The idea for this service is hard to get into without first knowing how some of the SR are working and there corner cases. Or at least we have to document it better.